core/src/main/java/tc/oc/pgm/structure/DynamicDefinition.java

```diff
@@ -0,0 +1,83 @@
+package tc.oc.pgm.structure;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import org.bukkit.util.Vector;
+import org.jetbrains.annotations.Nullable;
+import tc.oc.pgm.api.feature.FeatureInfo;
+import tc.oc.pgm.api.filter.Filter;
+import tc.oc.pgm.features.SelfIdentifyingFeatureDefinition;
+
+@FeatureInfo(name = "dynamic")
+public class DynamicDefinition extends SelfIdentifyingFeatureDefinition {
```

Comment:
```suggestion
@FeatureInfo(name = "dynamic-structure")
public class DynamicStructureDefinition extends SelfIdentifyingFeatureDefinition {
```
using dynamic as a global name is a bad idea, easy to confuse with dynamic filters